tests/unit/commands/test_init.py

```diff
@@ -14,36 +14,73 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
+import os
+import pathlib
 import sys
-from pathlib import Path
 from textwrap import dedent
-from unittest.mock import call
 
 import pytest
 
-from snapcraft import cli
+from snapcraft import application
 from snapcraft.models.project import Project
-from snapcraft.parts.yaml_utils import _SNAP_PROJECT_FILES, apply_yaml, process_yaml
-
-
-@pytest.fixture(autouse=True)
-def mock_argv(mocker):
-    return mocker.patch.object(sys, "argv", ["snapcraft", "init"])
-
-
-def test_init_default(emitter, new_dir):
+from snapcraft.parts.yaml_utils import apply_yaml, process_yaml
+
+
+@pytest.fixture
+def valid_new_dir(tmp_path):
+    """Change to a new temporary directory whose name is a valid snap name."""
+    cwd = os.getcwd()
+    new_dir = tmp_path / "test-snap-name-dir"
+    new_dir.mkdir()
+    os.chdir(new_dir)
+
+    yield new_dir
+
+    os.chdir(cwd)
```

Comment:
I started using `monkeypatch.chdir()` to avoid self-restoring working directory by myself, maybe we can incorporate that here as well?
```suggestion
def valid_new_dir(tmp_path, monkeypatch):
    """Change to a new temporary directory whose name is a valid snap name."""
    cwd = os.getcwd()
    new_dir = tmp_path / "test-snap-name-dir"
    new_dir.mkdir()
    monkeypatch.chdir(new_dir)
    return new_dir
```